src/mstch.cpp

```diff
@@ -478,55 +479,67 @@ ::mstch::node CXXRecord::methods()
     if (method_vector.size() > 0)
         method_vector.back()->setLast(true);
 
-    // Copy each template into the mstch template array.
-    for (auto method_template : method_vector)
-        method_templates.push_back(method_template);
+    // Collect all the non-static method names
+    std::unordered_set<std::string> non_static_method_names;
+    for (const std::shared_ptr<Method> &method : method_vector)
+    {
+        const bool is_static
+            = (::mstch::render("{{#is_static}}true{{/is_static}}", method)
+               == "true");
+
+        if (!is_static)
+            non_static_method_names.insert(method->nameAsString());
+    }
+
+    // Copy each template into the mstch template array except static methods
+    // that have the same name with instance methods.
+    for (const std::shared_ptr<Method> &method : method_vector)
+    {
+        const bool is_static
+            = (::mstch::render("{{#is_static}}true{{/is_static}}", method)
+               == "true");
+        if (is_static)
+        {
+            const std::string name = method->nameAsString();
+            if (non_static_method_names.find(name)
+                != non_static_method_names.end())
+            {
+                std::cerr
+                    << "Warning: Skipping static method '" << name
+                    << "' because there is instance method(s) with the same "
+                    << "name. Consider renaming the conflicting static method "
+                    << "in the configuration YAML.\n";
+                continue;
+            }
+        }
+
+        method_templates.push_back(method);
+    }
+
     return method_templates;
 }
 
 ::mstch::node CXXRecord::staticMethods()
 {
+    ::mstch::array static_method_list;
```

Comment:
Let's move this declaration to right above the `for (const ::mstch::node method_node : method_templates)` where it is used.

Reply:
Done